### PR TITLE
Feature/emoji on comment area

### PR DESCRIPTION
**Pull request by hydr063n**
##### Description

I created a feature that enables users to post emojis on their comment to an activity

##### Refers/Fixes

https://github.com/gitcoinco/web/issues/6035

##### Testing
Before:
https://imgur.com/a/x4EptXu

After:
https://streamable.com/ujhflw

### CODE REVIEW

File: app/assets/v2/css/activity_stream.css [hunk 1]
```diff
@@ -79,6 +79,10 @@ a .sup {
 .action.open {
   background-color: #eeeeee;
 }
+.emoji-container,
```
Comment (thelostone-mc) [inline app/assets/v2/css/activity_stream.css:82]: ^ could you add an empty line between CSS rules
(feel free if you've got the patience to do it for this css file

File: app/assets/v2/css/activity_stream.css [hunk 2]
```diff
@@ -79,6 +79,10 @@ a .sup {
...
+.comment-area{
```
Comment (thelostone-mc) [inline app/assets/v2/css/activity_stream.css:83]: Do we need set the `.comment-area` to `top:0` ?

File: app/assets/v2/js/activity.js [hunk 3]
```diff
@@ -739,8 +739,11 @@ $(document).ready(function() {
           <div class="col-sm-1 mt-1 activity-avatar d-none d-sm-inline">
             <img src="/dynamic/avatar/${document.contxt.github_handle}">
           </div>
-          <div class="col-12 col-sm-11 text-right">
+          <div class="comment-area col-12 col-sm-11 text-right">
             <textarea class="form-control bg-lightblue font-caption enter-activity-comment" placeholder="Enter comment" cols="80" rows="3">${existing_text}</textarea>
+            <div class="emoji-container position-absolute d-flex flex-wrap" style="width: 3.6em; right: 0; top: 0">
```
Comment (thelostone-mc) [inline app/assets/v2/js/activity.js:744]: could we shift the  inline styles to  `activity_stream.css`
and we can remove `top:0` cause it's already there in top

File: app/assets/v2/js/activity.js [hunk 4]
```diff
@@ -753,6 +756,24 @@ $(document).ready(function() {
...
+  });
+
+  picker.on('emoji', function(emoji) {
+    textArea.value += ` ${emoji} `;
+  });
+
+  $(document).on('click', '.emoji_button', function(e) {
+    e.preventDefault();
+    var commentArea = $(this).parents('.comment-area')[0];
+    var emojiContainer = $(this).parents('.emoji-container')[0];
```
Comment (thelostone-mc) [inline app/assets/v2/js/activity.js:772]: ^ const

File: app/assets/v2/css/activity_stream.css (hunk 1 above)
Comment (hydr063n) [reply app/assets/v2/css/activity_stream.css]: On the new commit, I have added an empty line and space between curly bracket for all the css rules for a more consistent code style

File: app/assets/v2/css/activity_stream.css (hunk 2 above)
Comment (hydr063n) [reply app/assets/v2/css/activity_stream.css]: No we didn't, I have since removed the rule and a few others that weren't necessary within inline css.

**Comment by hydr063n** [comment]
I've made the requested changes.

File: app/assets/v2/js/activity.js [hunk 5]
```diff
@@ -753,6 +758,24 @@ $(document).ready(function() {
     });
   };
 
+  // add emoji to comment
+  let textArea = '';
```
Comment (octavioamu) [inline app/assets/v2/js/activity.js:762]: how this code works when status is loaded? for example on profile activity.js and status.js are loaded and seems use the same var name.

Comment (hydr063n) [reply app/assets/v2/js/activity.js]: the code still works when both status.js and activity.js are loaded on the townsquare and profile page. I have searched for the var name and was unable to find a duplicate, however, i have renamed the variable name to a more precise name `activityCommentTextArea` 

screenshots: https://imgur.com/a/Upbk7M4

**Comment by hydr063n** [comment]
I have changed the duplicate variable name to a more precise name `activityCommentTextArea`

File: app/assets/v2/css/activity_stream.css [hunk 6]
```diff
@@ -5,26 +5,33 @@
   background-color: #ecf0f4;
```
Comment (danlipert) [inline app/assets/v2/css/activity_stream.css:5]: In general, please do not include arbitrary style changes in PRs as it makes it difficult to do a code review. Please limit your changes to the feature or fix thats described in the PR description - if you want to do a style fix PR, you can submit it separately.

**Comment by hydr063n** [comment]
I've made the requested changes. The emojis do have colors, I just had a terrible install of a Fedora Virtual Machine. Was pretty slow too, haha. I've since installed another flavor of Linux straight on the hdd. See video for color emojis https://streamable.com/cxrlp4

File: app/assets/v2/css/activity_stream.css (hunk 6 above)
Comment (hydr063n) [reply app/assets/v2/css/activity_stream.css]: I've removed the code style changes and left only changes I've made for the feature. I will follow this rule of thumb from now onwards.

**Comment by danlipert** [comment]
@hydr063n very good work, appreciate the fixes. classic linux lacking emoji support ;)

**Pull request merged**